Port ScrollingAccelerationCurve to the new IPC serialization format

cdumez opened the pull request:
#### 976ca2f077d0aef11a2d9ee26563f93bd7be1514
<pre>
Port ScrollingAccelerationCurve to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=264526">https://bugs.webkit.org/show_bug.cgi?id=264526</a>

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/ScrollingAccelerationCurve.cpp:
(WebKit::ScrollingAccelerationCurve::encode const): Deleted.
(WebKit::ScrollingAccelerationCurve::decode): Deleted.
* Source/WebKit/Shared/ScrollingAccelerationCurve.h:
(WebKit::ScrollingAccelerationCurve::gainLinear const):
(WebKit::ScrollingAccelerationCurve::gainParabolic const):
(WebKit::ScrollingAccelerationCurve::gainCubic const):
(WebKit::ScrollingAccelerationCurve::gainQuartic const):
(WebKit::ScrollingAccelerationCurve::tangentSpeedLinear const):
(WebKit::ScrollingAccelerationCurve::tangentSpeedParabolicRoot const):
(WebKit::ScrollingAccelerationCurve::resolution const):
* Source/WebKit/Shared/ScrollingAccelerationCurve.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/270499@main">https://commits.webkit.org/270499@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a6be18c88d66e8a32b7b914fcdae2d875eaeacd1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25639 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4244 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/26922 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27737 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23485 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/5991 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1679 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25888 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3160 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22096 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28319 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/2792 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/29132 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23397 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23412 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26984 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2809 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1047 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4184 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3253 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3277 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3125 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/ScrollingAccelerationCurve.serialization.in:
```diff
@@ -0,0 +1,36 @@
+#if ENABLE(MOMENTUM_EVENT_DISPATCHER)
+
+class WebKit::ScrollingAccelerationCurve {
+    float gainLinear();
```
You could've also exposed an accessor for Parameters and serialized it as a nested class, but then you would've also needed to add a constructor.  This is probably a tiny bit cleaner in this case.

cdumez committed:
Port ScrollingAccelerationCurve to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=264526

Reviewed by Alex Christensen.

* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Shared/ScrollingAccelerationCurve.cpp:
(WebKit::ScrollingAccelerationCurve::encode const): Deleted.
(WebKit::ScrollingAccelerationCurve::decode): Deleted.
* Source/WebKit/Shared/ScrollingAccelerationCurve.h:
(WebKit::ScrollingAccelerationCurve::gainLinear const):
(WebKit::ScrollingAccelerationCurve::gainParabolic const):
(WebKit::ScrollingAccelerationCurve::gainCubic const):
(WebKit::ScrollingAccelerationCurve::gainQuartic const):
(WebKit::ScrollingAccelerationCurve::tangentSpeedLinear const):
(WebKit::ScrollingAccelerationCurve::tangentSpeedParabolicRoot const):
(WebKit::ScrollingAccelerationCurve::resolution const):
* Source/WebKit/Shared/ScrollingAccelerationCurve.serialization.in: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/270499@main
